### PR TITLE
Improve the documentation about CPU tensors and named arguments

### PR DESCRIPTION
**Pull request by JanuszL**
- makes the note about the ability to accept only CPU tensors
  in named arguments more visible in getting started
- adds a relevant explanation why to the fn.transforms module
  describing why the operators there support only CPU as the device
- replaces CUDA 10.0 with 10.2 in the support matrix documentation section

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:

**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- makes the note about the ability to accept only CPU tensors
  in named arguments more visible in getting started
- adds a relevant explanation why to the fn.transforms module
  describing why the operators there support only CPU as the device
- replaces CUDA 10.0 with 10.2 in the support matrix documentation section

### Affected modules and functionalities:
- getting started
- operators description page
- support matrix documentation page

### Key points relevant for the review:
- NA

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [X] N/A

### Documentation
- [ ] Existing documentation applies
- [X] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [X] RST
  - [X] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2550, DALI-2551

### CODE REVIEW

File: docs/autodoc_submodules.py [hunk 1]
```diff
@@ -24,6 +23,12 @@
 exclude_fn_members = {
 }
 
+mod_aditional_doc = {
+    'nvidia.dali.fn.transforms' : "All operators in this module supports only CPU device as they are meant " +
```
Comment (jantonguirao) [inline docs/autodoc_submodules.py:27]:
```suggestion
    'nvidia.dali.fn.transforms' : "All operators in this module support only CPU device as they are meant " +
```

File: docs/autodoc_submodules.py [hunk 2]
```diff
@@ -24,6 +23,12 @@
...
+"to be provided as an input to named operator arguments. Check for more details the relevant " +
```
Comment (klecki) [inline docs/autodoc_submodules.py:28]: I think we should try to stick with one, consistent naming.
We started using the "argument input" term that we use in the backend, and explain it in the Python layer as named keyword arguments. Please consider using one of those.

File: docs/examples/getting_started.ipynb [hunk 3]
```diff
@@ -419,7 +419,13 @@
...
     "Rotate's `angle` parameter can accept `float` or `float tensor` types of values. The second option, `float tensor`, enables us to feed the operator with different rotation angles for every image, via a tensor produced by other operation.\n",
     "\n",
-    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle."
+    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle.\n",
+    "\n",
+    "> **_NOTE_** Keep in mind, that every time you pass an oputput of one DALI operator to another as named keyword argument (like the angle for the rotate above) that data must be placed on the CPU (the random.uniform has the default device='cpu').\n",
+    ">\n",
+    "> Only the regular inputs (the non-named positional ones) can use either CPU or GPU data, as show below.\n",
+    ">\n",
+    "> More information can be found in the pipeline documentation section"
```
Comment (klecki) [inline docs/examples/getting_started.ipynb:428]: Please add the link to appropriate section of the documentation.

File: docs/examples/getting_started.ipynb [hunk 4]
```diff
@@ -419,7 +419,13 @@
     "\n",
     "Rotate's `angle` parameter can accept `float` or `float tensor` types of values. The second option, `float tensor`, enables us to feed the operator with different rotation angles for every image, via a tensor produced by other operation.\n",
     "\n",
-    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle."
+    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle.\n",
+    "\n",
+    "> **_NOTE_** Keep in mind, that every time you pass an oputput of one DALI operator to another as named keyword argument (like the angle for the rotate above) that data must be placed on the CPU (the random.uniform has the default device='cpu').\n",
+    ">\n",
+    "> Only the regular inputs (the non-named positional ones) can use either CPU or GPU data, as show below.\n",
```
Comment (jantonguirao) [inline docs/examples/getting_started.ipynb:426]:
```suggestion
    "> Only regular inputs (non-named, positional ones) can use either CPU or GPU data, as show below.\n",
```

File: docs/examples/getting_started.ipynb [hunk 5]
```diff
@@ -419,7 +419,13 @@
     "\n",
     "Rotate's `angle` parameter can accept `float` or `float tensor` types of values. The second option, `float tensor`, enables us to feed the operator with different rotation angles for every image, via a tensor produced by other operation.\n",
     "\n",
-    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle."
+    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle.\n",
+    "\n",
+    "> **_NOTE_** Keep in mind, that every time you pass an oputput of one DALI operator to another as named keyword argument (like the angle for the rotate above) that data must be placed on the CPU (the random.uniform has the default device='cpu').\n",
```
Comment (klecki) [inline docs/examples/getting_started.ipynb:424]: Wasn't it better to have it as a dedicated section?

Also, you can consider having a small code block with the mentioned code in this section, and mention the name of the variable that we pass as the argument input.

Comment (JanuszL) [reply docs/examples/getting_started.ipynb:424]: I think it was easy to miss it before. Now this is in one paragraph.
Also I think the code is already described:
> (like the angle for the rotate in below example) that data must be placed on the CPU (the random.uniform has the default device='cpu').

Comment (klecki) [reply docs/examples/getting_started.ipynb:424]: Ok, I agree that the placement is better, I'm not convinced about the formatting - it's not like we have warning box in notebooks, that why I was talking about dedicated section. I didn't notice how far up you moved it.

Now it references code "above", and the code is below.

File: docs/autodoc_submodules.py (hunk 2 above)
Comment (JanuszL) [reply docs/autodoc_submodules.py]: Done

File: docs/examples/getting_started.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/getting_started.ipynb:424]: Done

File: docs/examples/getting_started.ipynb (hunk 4 above)
Comment (JanuszL) [reply docs/examples/getting_started.ipynb:426]: Done

File: docs/examples/getting_started.ipynb (hunk 5 above)
Comment (JanuszL) [reply docs/examples/getting_started.ipynb:424]: We don't, this is the closest thing I was able to come up (I'm open to suggestions). I think it deserves a highlight.

**Review by jantonguirao — APPROVED** [review]

File: docs/examples/getting_started.ipynb (hunk 5 above)
Comment (JanuszL) [reply docs/examples/getting_started.ipynb:424]: adjusted `above->below`

**Comment by JanuszL** [comment]
!build

File: docs/examples/getting_started.ipynb [hunk 6]
```diff
@@ -419,7 +419,15 @@
...
-    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle."
+    "Random number generators are examples of operations that one can use with DALI. Let us use `fn.random.uniform` to make a pipeline that rotates images by a random angle.\n",
+    "\n",
+    "> **_NOTE_**\n",
+    ">\n",
+    "> Keep in mind, that every time you pass an output of a DALI operator to another as a named keyword argument (like the angle for the rotate above) that data must be placed on the CPU (``random.uniform`` has a default device='cpu').\n",
+    ">\n",
+    "> Only regular inputs (non-named, positional ones) can use either CPU or GPU data, as show below.\n",
+    ">\n",
+    "> More information can be found in the [pipeline documentation section](../pipeline.html?#processing-graph-structure)."
```
Comment (klecki) [inline docs/examples/getting_started.ipynb:430]:
```suggestion
    "> Keep in mind, that every time you pass an output of a DALI operator to another as a named keyword argument that data must be placed on the CPU. In the example below, we use the output of ``random.uniform`` (which has a default device='cpu') as ``angle`` keyword argument to ``rotate``.\n",
    ">\n",
    "> Such arguments in DALI are called "argument inputs", .more information about them can be found in the [pipeline documentation section](../pipeline.html?#processing-graph-structure)."
    "> Regular inputs (non-named, positional ones) do not have such constraints and can use either CPU or GPU data, as show below.\n",
    ">\n",
```
The part about them being argument inputs is not necessary, but I think the first sentence will be ordered better with what I propose.

Comment (JanuszL) [reply docs/examples/getting_started.ipynb]: Fixed

**Review by klecki — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Pull request merged**